src/init/pkg_installer.sh

```diff
@@ -1,194 +1,67 @@
-function restore_ossec_ownership {
-        while read -r line; do
-            TFILE_OWN=$(echo $line | cut -d " " -f -1)
-            TFILE_PATH=$(echo $line | cut -d " " -f 2-)
-            chown $TFILE_OWN "${TFILE_PATH}"
-        done <<< "$OSSEC_LIST_FILES"
+OS=$(uname)
 
-        # If the list is not empty, then it is assumed that the version to restore is lower than 4.3
-        if [ -n "${OSSEC_LIST_FILES}" ]; then
-            # If there are files with the group or user wazuh, then they are changed to ossec.
-            find ./ -group wazuh -exec chgrp ossec {} \;
-            find ./ -user wazuh -exec chown ossec {} \;
-            # Delete user and group
-            if command -v deluser > /dev/null 2>&1; then
-               deluser wazuh > /dev/null 2>&1
-            else
-               userdel wazuh > /dev/null 2>&1
-            fi
-            # Delete user and group
-            if command -v delgroup > /dev/null 2>&1; then
-               delgroup wazuh > /dev/null 2>&1
-            else
-               groupdel wazuh >/dev/null 2>&1
-            fi
+if [[ "$OS" == "Darwin" ]]; then
+    installer -pkg ./var/upgrade/wazuh-agent* -target / >> ./logs/upgrade.log 2>&1
+elif [[ "$OS" == "Linux" ]]; then
+    if find ./var/upgrade/ -mindepth 1 -maxdepth 1 -type f -name "*.rpm" | read; then
+        if command -v rpm >/dev/null 2>&1; then
+            rpm -Uvh ./var/upgrade/wazuh-agent* >> ./logs/upgrade.log 2>&1
+        else
+            echo "$(date +"%Y/%m/%d %H:%M:%S") - Upgrade failed. RPM package found but rpm command not found." >> ./logs/upgrade.log
+            echo -ne "2" > ./var/upgrade/upgrade_result
+            rm -f $LOCK
+            exit 1
         fi
-}
-
-# Restore SELinuxPolicy
-function restore_selinux_policy {
-    if command -v semodule > /dev/null && command -v getenforce > /dev/null; then
-        if [ $(getenforce) != "Disabled" ]; then
-            if [ -f ./var/selinux/wazuh.pp ]; then
-                echo "$(date +"%Y/%m/%d %H:%M:%S") - Restoring SELinux policy." >> ./logs/upgrade.log
-                semodule -i ./var/selinux/wazuh.pp >> ./logs/upgrade.log 2>&1
-                semodule -e wazuh >> ./logs/upgrade.log 2>&1
-            else
-                echo "$(date +"%Y/%m/%d %H:%M:%S") - ERROR: Wazuh SELinux module not found." >> ./logs/upgrade.log
-            fi
+    elif find ./var/upgrade/ -mindepth 1 -maxdepth 1 -type f -name "*.deb" | read; then
+        if command -v dpkg >/dev/null 2>&1; then
+            dpkg -i ./var/upgrade/wazuh-agent* >> ./logs/upgrade.log 2>&1
         else
-            echo "$(date +"%Y/%m/%d %H:%M:%S") - Wazuh SELinux module installation skipped (SELinux is disabled)." >> ./logs/upgrade.log
+            echo "$(date +"%Y/%m/%d %H:%M:%S") - Upgrade failed. DEB package found but dpkg command not found." >> ./logs/upgrade.log
+            echo -ne "2" > ./var/upgrade/upgrade_result
+            rm -f $LOCK
+            exit 1
+        fi
+    else
```

Comment:
Check https://documentation.wazuh.com/current/installation-guide/wazuh-agent/wazuh-agent-package-linux.html. Be aware we should also support `apk` packages.

Reply:
Added in 08f1fbde55339e4fad0565a97f8399e464d40bd9